Add a new WebTool: AI Chat WebTool

esatiyev opened the pull request:
This pull request contains the current working version of the AI Chat WebTool being developed under Google Summer of Code 2025.

The implementation builds upon prior work by @rmackay9 and other contributors. My additions are integrated with that foundation and reflect the state of development so far.

All changes have been tested locally. Feedback is welcome.


<img width="1843" height="1368" alt="Screenshot from 2025-08-07 11-55-51" src="https://github.com/user-attachments/assets/7505e65a-8e51-49df-bb37-38867e6c2aa8" />

khanasif786 commented:
@esatiyev We can add tool in the Development section "Work is Progress Tools"

esatiyev commented:
@khanasif786 Done, I added it to the "Work in Progress" section.

esatiyev commented:
@khanasif786 Done, thanks

rmackay9 reviewed (approved):
Thanks very much for this!

IamPete1 commented:
There is a bit of MAVLink re-work in https://github.com/ArduPilot/WebTools/pull/267 too. It would be good if they could converge to keep the lib  in the same place and use it in the same way.

IamPete1 reviewed (changes requested):
I think `Chat.js` and `mavlink_msgs.js` can just be removed?

esatiyev commented:
> There is a bit of MAVLink re-work in #267 too. It would be good if they could converge to keep the lib in the same place and use it in the same way.

I’ve moved MAVLink to modules, same as in PR #267, and updated my index.html file to use it.
